ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -309,9 +312,6 @@ cacheDataProducer uri packageState curMod globalEnv inScopeEnv limports deps = d
       -- Full canonical names of imported modules
       importDeclerations = map unLoc limports
 
-      -- The list of all importable Modules from all packages
-      moduleNames = map showModName (listVisibleModuleNames dflags)
-
```

Comment:
Since this is a constant and it depends only on the session itself, I think it would make sense to cache it and compute only once, with the necessary error handling. 

The best way to cache things that depend only on the GHC session is to add them to the HscEnvEq record. 

@berberman could you investigate that approach?